jenkins/version-def.sh

```diff
@@ -48,9 +60,28 @@ SPARK_REPO=${SPARK_REPO:-"$URM_URL"}
 echo "CUDF_VER: $CUDF_VER, CUDA_CLASSIFIER: $CUDA_CLASSIFIER, PROJECT_VER: $PROJECT_VER \
     SPARK_VER: $SPARK_VER, SCALA_BINARY_VER: $SCALA_BINARY_VER"
 
+# Spark shim versions
+. $(dirname "$0")/common.sh
+case $PHASE_TYPE in
+    pre-release)
+        PROFILE_OPT="-Psnapshots" 
+        ;;
 
-SPARK_SHIM_VERSIONS_STR=${SPARK_SHIM_VERSIONS_STR:-"311 321cdh 312 313 314 320 321 322 330 331"}
-
-IFS=" " <<< $SPARK_SHIM_VERSIONS_STR read -r -a SPARK_SHIM_VERSIONS
-
+    *)
+        PROFILE_OPT="-PnoSnapshots" 
+        ;;
+esac
+# base version
+get_spark_shim_versions $PROFILE_OPT
 SPARK_BASE_SHIM_VERSION=${SPARK_SHIM_VERSIONS[0]}
+# snapshots + noSnapshots
+get_spark_shim_versions -Psnapshots
+SPARK_SHIM_VERSIONS_ALL=("${SPARK_SHIM_VERSIONS[@]}")
```

Comment:
I don't think SPARK_BASE_VERSION will change between `get_spark_shim_versions $PROFILE_OPT` and `get_spark_shim_versions -Psnapshots` so we can skip  the first call to get_spark_shim_versions

```suggestion
# snapshots + noSnapshots
get_spark_shim_versions -Psnapshots
SPARK_BASE_SHIM_VERSION="${SPARK_SHIM_VERSIONS[0]}"
SPARK_SHIM_VERSIONS_ALL=("${SPARK_SHIM_VERSIONS[@]}")
```

Reply:
For current pom profiles, `get_spark_shim_versions $PROFILE_OPT` and `get_spark_shim_versions -Psnapshots` will return same result.
I'm not sure if we will change the first element of `snapshots` or `noSnapshots` profile in pom. @pxLi

Reply:
snapshots and noSnapshots should always share the same default spark shim (currently 311), if we decide to change it the first element should also be the same~